File: frontend/lib/src/data/model/goal/goal_model.dart
import 'package:orizon/src/domain/entities/goal/goal_entity.dart';

class GoalModel extends GoalEntity {
  const GoalModel({
    required super.id,
    required super.title,
    required super.targetAmount,
    required super.currentAmount,
    super.deadline,
    required super.status,
    required super.createdAt,
  });

  factory GoalModel.fromJson(Map<String, dynamic> json) {
    return GoalModel(
      id: json['id'],
      title: json['title'],
      targetAmount: (json['targetAmount'] as num).toDouble(),
      currentAmount: (json['currentAmount'] as num).toDouble(),
      deadline: json['deadline'] != null ? DateTime.parse(json['deadline']) : null,
      status: GoalStatus.values.firstWhere(
        (s) => s.name.toUpperCase() == json['status'],
        orElse: () => GoalStatus.active,
      ),
      createdAt: DateTime.parse(json['createdAt']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'targetAmount': targetAmount,
      'deadline': deadline?.toIso8601String().split('T').first,
    };
  }
}

File: frontend/web/src/app/core/models/goal.model.ts
export type GoalStatus = 'ACTIVE' | 'COMPLETED' | 'CANCELLED';

export interface GoalResponse {
  id: string;
  title: string;
  targetAmount: number;
  currentAmount: number;
  deadline: string | null;
  status: GoalStatus;
  createdAt: string;
}

export interface GoalContributionResponse {
  id: string;
  goalId: string;
  amount: number;
  date: string;
  createdAt: string;
}

export interface CreateGoalRequest {
  title: string;
  targetAmount: number;
  deadline?: string;
}

export interface UpdateGoalRequest {
  title: string;
  targetAmount: number;
  deadline?: string;
}

export interface CreateGoalContributionRequest {
  amount: number;
  date: string;
}
